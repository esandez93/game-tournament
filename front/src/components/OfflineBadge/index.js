import OfflineBadge from './OfflineBadge';

export default OfflineBadge;