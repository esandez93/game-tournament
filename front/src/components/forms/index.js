import UserInfo from './UserInfo';
import Settings from './Settings';

// TODO: Refactor UserInfo and Settings into using Form component
export {
  UserInfo,
  Settings
};
