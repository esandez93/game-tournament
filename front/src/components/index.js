import Avatar from './Avatar';
import Button from './Button';
import Card from './Card';
import CharacterAvatar from './CharacterAvatar';
import ContextMenu from './ContextMenu';
import Form from './Form';
import Header from './Header';
import Input from './Input';
import LoadableLoading from './LoadableLoading';
import Loading from './Loading';
import Match from './Match';
import OfflineBadge from './OfflineBadge';
import Select from './Select';
import SideMenu from './SideMenu';
import Stat from './Stat';
import Snackbar from './Snackbar';
import Switch from './Switch';
import Tabs from './Tabs';

export {
  Avatar,
  Button,
  Card,
  CharacterAvatar,
  ContextMenu,
  Form,
  Header,
  Input,
  LoadableLoading,
  Loading,
  Match,
  OfflineBadge,
  Select,
  SideMenu,
  Stat,
  Snackbar,
  Switch,
  Tabs
};
