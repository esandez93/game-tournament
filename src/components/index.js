import Avatar from './Avatar';
import Button from './Button';
import Icon from './Icon';
import Loading from './Loading';
import OfflineBadge from './OfflineBadge';
import SideMenu from './SideMenu';
import Snackbar from './Snackbar';

export {
  Avatar,
  Button,
  Icon,
  Loading,
  OfflineBadge,
  SideMenu,
  Snackbar
};
