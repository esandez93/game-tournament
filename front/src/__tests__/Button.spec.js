import React from "react";
import ReactDOM from "react-dom";
import { act } from "react-dom/test-utils";

import { Button } from '@/components';

let container;

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  document.body.removeChild(container);
  container = null;
});

describe("Button component", () => {
  it('should render without crashing', () => {
    act(() => {
      ReactDOM.render(<Button>Test</Button>, container);
    });
  });

  it('should have expected content', () => {
    const textContent = 'Test';

    act(() => {
      ReactDOM.render(<Button>{textContent}</Button>, container);
    });

    const button = container.querySelector('button');
    expect(button.textContent).toBe(textContent);
  });
});
