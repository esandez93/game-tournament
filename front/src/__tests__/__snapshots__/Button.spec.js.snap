// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render without crashing 1`] = `
<WithStyles(ForwardRef(Button))
  className="Button"
  color="default"
  variant="contained"
/>
`;
